tiledb/sm/array_schema/array_schema.cc

```diff
@@ -473,21 +443,7 @@ Status ArraySchema::add_attribute(const Attribute* attr, bool check_special) {
   }
 
   // Create new attribute and potentially set a default name
-  auto new_attr = (Attribute*)nullptr;
-  if (attr->is_anonymous()) {
-    // Check if any other attributes are anonymous
-    for (auto& a : attributes_) {
-      if (a->is_anonymous()) {
-        return LOG_STATUS(Status::ArraySchemaError(
-            "Only one anonymous attribute is allowed per array"));
-      }
-    }
-    new_attr = new Attribute(attr);
-    new_attr->set_name(constants::default_attr_name);
-  } else {
-    new_attr = new Attribute(attr);
-  }
-
+  auto new_attr = new Attribute(attr);
```

Comment:
Not specific to your change, but I think the `Attribute` instances we store in `attributes_`/`attribute_map_` are needlessly heap-allocated. If you think an end-user will ever store thousands of attributes, I'll create a ticket to refactor this into stack-allocated `Attribute` instances.

Reply:
Yeah, pointers make it easier than std references here for the map. There will never be a case where the attributes are in the order of thousands. If it ever comes to that we can certainly revisit.